microcosm_postgres/factories/engine_strategy.py

```diff
@@ -0,0 +1,54 @@
+from microcosm.scoping.proxies import ScopedProxy
+
+
+def unwrap(bind):
+    return bind.__component__ if isinstance(bind, ScopedProxy) else bind
+
+
+class EngineRoutingStrategy:
+    """
+    A pluggable strategy for use in our `RoutingSession`.
+
+    """
+    def __init__(self, graph):
+        self.graph = graph
+
+    def get_bind(self, mapper, clause=None):
+        """
+        Implement `Session.get_bind`
+
+        """
+        return unwrap(self.graph.postgres)
+
+    @property
+    def supports_multiple_binds(self):
+        """
+        Does this strategy support multiple concurrent engine binds?
+
+        """
+        return False
+
+
+class ModelEngineRoutingStrategy:
```

Comment:
This is the logic I used previously for my tests.

I'm not sure we'll ever use it, but I need an example.